File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int st = 0;
        int lst = n-1;
        int ans = 0;
        while(st<lst){
            int h = min(height[st],height[lst]);
            int w = lst-st;
            int ar = h*w;
            if(ans < ar){
                ans= ar;
            }
            if(height[st] < height[lst]){
                st++;
            }
            else{
                lst--;
            }
        }
        return ans;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int st = 0;
        int end = n-1;
        int maxtw = 0;
        while(st < end){
            int h = Math.min(height[st], height[end]);
            int dis = end-st;
            int curr = h*dis;
            maxtw = Math.max(curr,maxtw);
            if(height[st] < height[end]){
                st++;
            }
            else{
                end--;
            }   
        }
        return maxtw;   
    }
}